system_test/aest_hard_fork_SUITE.erl

```diff
@@ -0,0 +1,564 @@
+          mine_rate => default,
+          hard_forks => ?PROTOCOLS(H)
+         }).
+
+-define(NEW_NODE2(H), #{
+          name    => new_node2,
+          peers   => [new_node1],
+          backend => aest_docker,
+          source  => {pull, "aeternity/epoch:local"},
+          mine_rate => default,
+          hard_forks => ?PROTOCOLS(H)
+         }).
+
+-define(NEW_NODE3(H), #{
+          name    => new_node3,
+          peers   => [new_node1],
+          backend => aest_docker,
+          source  => {pull, "aeternity/epoch:local"},
+          mine_rate => default,
+          hard_forks => ?PROTOCOLS(H)
+         }).
+
+-define(NEW_NODE4(H), #{
+          name    => new_node4,
+          peers   => [new_node3],
+          backend => aest_docker,
+          source  => {pull, "aeternity/epoch:local"},
+          mine_rate => default,
+          hard_forks => ?PROTOCOLS(H)
+         }).
+
+-define(FAST_NEW_NODE1(H), #{
+          name    => fast_new_node1,
+          peers   => [],
+          backend => aest_docker,
+          source  => {pull, "aeternity/epoch:local"},
+          mine_rate => 1000,
+          cuckoo_miner => ?CUCKOO_MINER(16),
+          hard_forks => ?PROTOCOLS(H)
+         }).
+
+-define(FAST_NEW_NODE2(H), #{
+          name    => fast_new_node2,
+          peers   => [fast_new_node1],
+          backend => aest_docker,
+          source  => {pull, "aeternity/epoch:local"},
+          mine_rate => 1000,
+          cuckoo_miner => ?CUCKOO_MINER(16),
+          hard_forks => ?PROTOCOLS(H)
+         }).
+
+%=== COMMON TEST FUNCTIONS =====================================================
+
+all() ->
+    [
+     {group, assumptions},
+     {group, hard_fork}
+    ].
+
+groups() ->
+    [
+     {assumptions,
+      [
+       {genesis,
+        [sequence], %% Hard deps among tests.
+        [
+         old_node_persisting_chain_and_not_mining_has_genesis_as_top,
+         new_node_persisting_chain_and_not_mining_has_same_old_genesis_as_top
+        ]},
+       new_nodes_can_mine_and_sync_fast_minimal_chain_with_pow
+      ]},
+     {hard_fork,
+      [sequence], %% Hard deps among tests/groups.
+      [
+       restore_db_backup_on_old_node, %% Determines info of top of chain in DB backup.
+       {group, hard_fork_all}
+      ]},
+     {hard_fork_all,
+      [sequence], %% Soft deps among tests/groups: if a test/group fails better skipping the rest.
+      [
+       old_node_can_receive_chain_from_other_old_node,
+       {group, upgrade_flow_smoke_test}
+      ]},
+     {upgrade_flow_smoke_test,
+      [sequence], %% Hard deps among tests/groups.
+      [
+       restore_db_backup_with_short_chain_on_new_node,
+       new_node_accepts_long_old_chain_from_other_new_node_up_to_height_of_new_protocol,
+       new_node_can_receive_short_old_chain_from_other_new_node,
+       new_node_can_mine_on_old_chain_using_old_protocol,
+       new_node_can_mine_on_old_chain_using_new_protocol
+      ]}
+    ].
```

Comment:
I think the indentation of the groups here is very unreadable. Perhaps you could indent one tab stop or keep starts of lists and the first tuple on the same line (e.g. `[{assumptions,\n`)

Reply:
I plan not to address. A coding style would be beneficial - yes.